### PR TITLE
Tape recorder soulefaction (most unatomised PR ever)

### PR DESCRIPTION
**Pull request by stanalbatross**
# About the pull request

updates the tape recorder to be nicer to use + have a bit more soul to it (imported from tg)

and a ton of backend stuff

# Explain why it's good for the game

makes the recorder cooler + backend is cool

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: the tape recorder now uses abovehead chat too
add: the tape recorder now uses external tapes that can be removed. Tape recorder tapes can be flipped over to record on the other side. They can also be renamed with pens. You can unspool them by using their radial menu too. Sprites for open tape recorder by nauticall.
add: ported a bunch of TG sounds for the tape recorder
ui: the tape recorder now uses radial menus.
refactor: refactored labels to use a component
refactor: pens now use a tooltrait rather than istype checks
code: changed a lot of language lists to use list defines
add: pens now have the ability to modify the base names and descriptions of certain items... currently only the tape recorder tapes but support for it is there!
add: added sounds for writing on paper
/:cl:

### CODE REVIEW

**Review by carlarctg — COMMENTED** [review]
erm thr code goody

File: code/modules/projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -1121,7 +1121,6 @@ obj/item/weapon/gun/launcher/grenade/update_icon()
 
 /obj/item/weapon/gun/launcher/rocket/anti_tank/disposable/proc/fold(mob/user)
 	var/obj/item/prop/folded_anti_tank_sadar/F = new /obj/item/prop/folded_anti_tank_sadar(src.loc)
-	F.set_name_label(name_label)
```
Comment (Diegoflores31) [inline code/modules/projectiles/guns/specialist.dm:1124]: Transfer_label_component()

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -1158,7 +1157,6 @@ obj/item/weapon/gun/launcher/grenade/update_icon()
 
 /obj/item/prop/folded_anti_tank_sadar/proc/unfold(mob/user)
 	var/obj/item/weapon/gun/launcher/rocket/anti_tank/disposable/F = new /obj/item/weapon/gun/launcher/rocket/anti_tank/disposable(src.loc)
-	F.set_name_label(name_label)
```
Comment (Diegoflores31) [inline code/modules/projectiles/guns/specialist.dm:1161]: Transfer_label_component()

File: code/game/objects/items/devices/taperecorder.dm [hunk 3]
```diff
@@ -1,207 +1,492 @@
...
 	set category = "Object"
-	set src in usr
-	if(usr.stat)
+
+	if(!mytape.storedinfo.len)
+		return
+	if(!can_use(usr))
+		return
+	if(!mytape)
 		return
```
[the remark this answers is not in the document]
Comment (stanalbatross) [reply code/game/objects/items/devices/taperecorder.dm:315]: it's a bool

File: code/game/objects/items/devices/taperecorder.dm [hunk 4]
```diff
@@ -1,207 +1,492 @@
...
+	else if(icon_state == "cassette_flip") //so flipping doesn't overwrite an unexpected icon_state (e.g. an admin's)
+		icon_state = initial_icon_state
+
+//Random colour tapes
+/obj/item/tape/random
+	icon_state = "random_tape"
+
+var/list/cassette_colours = list("blue", "gray", "green", "orange", "pink_stripe", "purple", "rainbow", "red_black", "red_stripe", "camo", "rising_sun", "orange", "blue", "ocean", "aesthetic")
+
+var/list/cassette_map_themes = list("solaris", "ice", "lz", "dam", "worstmap")
```
[the remark this answers is not in the document]
Comment (stanalbatross) [reply code/game/objects/items/devices/taperecorder.dm:484]: for later

File: code/game/sound.dm [hunk 5]
```diff
@@ -231,6 +231,8 @@
 				S = pick('sound/effects/pry1.ogg', 'sound/effects/pry2.ogg','sound/effects/pry3.ogg','sound/effects/pry4.ogg')
 			if("metalbang")
 				S = pick('sound/effects/thud1.ogg','sound/effects/thud2.ogg','sound/effects/thud3.ogg')
+			if("paper_writing")
+				S = pick('sound/items/writing_noises/paper_writing_1.wav', 'sound/items/writing_noises/paper_writing_2.wav', 'sound/items/writing_noises/paper_writing_3.wav', 'sound/items/writing_noises/paper_writing_4.ogg')
 			// Weapons/bullets
```
[the remark this answers is not in the document]
Comment (stanalbatross) [reply code/game/sound.dm:236]: yes, deal with it

File: code/game/objects/items/devices/taperecorder.dm [hunk 6]
```diff
@@ -1,207 +1,492 @@
...
+	pickup_sound = 'sound/handling/taperecorder_pickup.ogg'
+	flags_atom = FPRINT|CONDUCT|USES_HEARING
+
+	var/recording = FALSE
+	var/playing = FALSE
+	var/playsleepseconds = 0
+	var/obj/item/tape/mytape
+	var/starting_tape_type = /obj/item/tape/regulation
+	var/canprint = TRUE
+	var/list/known_languages = list(LANGUAGE_ENGLISH, LANGUAGE_JAPANESE, LANGUAGE_RUSSIAN, LANGUAGE_GERMAN, LANGUAGE_SPANISH, LANGUAGE_CHINESE)
```
[the remark this answers is not in the document]
Comment (stanalbatross) [reply code/game/objects/items/devices/taperecorder.dm:24]: yes it should be

**Comment by fira** [comment]
Looks fine at a glance (althoguh that horrible hands proc alias is making me cry) 
Might want to TM this due to scope with the backend changes

File: code/game/objects/items/devices/taperecorder.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/game/objects/items/devices/taperecorder.dm:319]: print_on_cooldown then smgh

**Review by harryob — APPROVED** [review]

**Comment by Hopekz** [comment]
Tape recorder radial and other ported out of my PR's here: tgstation/tgstation#51653 and Yogstation13/yogstation#8446

**Pull request merged**